src/checkers/inference/solver/util/ConstantSlotUtils.java

```diff
@@ -0,0 +1,57 @@
+package checkers.inference.solver.util;
+
+import org.checkerframework.javacutil.AnnotationUtils;
+
+import java.util.Collection;
+
+import javax.lang.model.element.AnnotationMirror;
+
+import checkers.inference.model.ComparableConstraint;
+import checkers.inference.model.ConstantSlot;
+import checkers.inference.model.Constraint;
+import checkers.inference.model.EqualityConstraint;
+import checkers.inference.model.InequalityConstraint;
+import checkers.inference.model.SubtypeConstraint;
+import checkers.inference.solver.frontend.Lattice;
+
+/**
+ * Utils methods for constant slot.
+ * 
+ * @author jianchu
+ *
+ */
+public class ConstantSlotUtils {
+
+    public static boolean checkConstant(ConstantSlot constant1, ConstantSlot constant2,
```

Comment:
This method is actually for early checking constraints that consist of constant slots.

Provide javadoc on this method. Describe the purpose, and think of a better method name.

`checkConstant` is confusing.

Reply:
Currently nobody use this method. Also I don't quite like the design of this method because:

1) Lack of argument verification: client actually can passing two arbitrary constant slots into this method, it lack of the check about if these passed slots were actually belongs to the given constraint.

2)Mis-leading response for un-handle cases: For the type of constraint it doesn't handle, it will directly return `true`, which might be mis-leading.

3) Missing report reasoning: Usually for early checking, at least we want to know some basic reasoning of why we think a given constraint failed the early verification. However this method just return a boolean value without any reasoning.

I would first leave/remove this method in this PR, and do a separate PR for adding early verification about constant constraints.